### DISCUSSION TITLE
Shop speed optimisation

### DISCUSSION THREAD

**Discussion by paraskumar12011996** [FriendsOfShopware/FroshPlatformThumbnailProcessor#139] [Q&A] [answered]
Hello Friends,
I have observed the Shopware Extension “ThumbnailProcessor incl” as root cause for page speed reduction.
If this plugin is active light house performance for mobile is 29 and desktop is 58
If I deactivate this plugin and verify the light house performance for mobile is 41 and desktop is 73

So can you please suggest me some ways and ideas to improve the speed with this plugin.

Thank you.

**Comment by tinect** [comment]
what service are you using for live-generation?

**Reply by paraskumar12011996** [reply]
You means service provider name which responsible for generation and deliver thumbnail real time when visiting shop?

**Reply by tinect** [reply]
yes. you can find some here: https://github.com/FriendsOfShopware/FroshPlatformThumbnailProcessor/discussions/categories/patterns

**Reply by paraskumar12011996** [reply]
hi @tinect ,
Gm!

So this plugin only responsible for generating the thumbnail URL, and service provider will be responsible for deliver it on right time when user visit the page.

So this speed is vary and depends on service provider. am I right?

Thank you

**Reply by tinect** [reply, marked as answer]
Hey @paraskumar12011996 

yes, here is the extract of the store description:

> Please note that this plugin only provides the function for delivering the thumbnail urls.
This plugin does not create thumbnails! The corresponding services are used for this.

**Reply by parasdhaduk9010** [reply]
Hello @tinect,
Hope you are fine and doing well.

I am using cloudimage for optimisation and also refereed (https://github.com/FriendsOfShopware/FroshPlatformThumbnailProcessor/discussions/39) discussion.

Can you please explain that the {mediaPath}  and {width} parameter 
**For example** 
media/7a/5b/4a/1710408151/example_1920x500px_one_v2.jpg
media/00/11/b0/1710407949/example_two_1920x480_v3.jpg
media/6e/a9/63/1706716095/example_three_1920x480.jpg

So after media their is dynamic path for media images So what will be the right media path for configuration and my thumbnail pattern. Please guide here.
![image](https://github.com/user-attachments/assets/e05f26b4-56a4-4567-afc3-f850f5865041)

**Reply by tinect** [reply]
There is no guide neccessary, just use these as placeholders. And check the examples in your posted screenshot.

**Reply by paraskumar12011996** [reply]
@tinect, Got it.

Thank you so much for your quick response and support.
